### PR TITLE
Add automated release process

### LINKED ISSUE #18: Add release process
We need an automated release process for doing ponyup releases.

It should:

* automatically update changelog ala ponyc
* automatically update release notes as we do for libraries
* create the release docker image
* create and upload release artifact to CloudSmith

Ideally, this would all be automated using tags as we do with libraries.

### PR DESCRIPTION
**Pull request by SeanTAllen**
The release process will create a binary and upload it to cloudsmith.
It also updates the CHANGELOG file, updates the VERSION, fills out
release notes, posts a notice to LWIP, and to the Zulip.

Closes #18

### CODE REVIEW

**Comment by SeanTAllen** [comment]
@Theodus I had to update the bootstrap to reflect the change in the nightly. Otherwise, the tests wouldn't pass.

**Comment by SeanTAllen** [comment]
@Theodus @EpicEric I know neither of you is very familiar with the automated release stuff but eyes for "I think this looks wrong" would be helpful.

File: RELEASE_PROCESS.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# How to cut a ponyup release
+
+This document is aimed at members of the team who might be cutting a release of of ponyup. It serves as a checklist that can take you through doing a release step-by-step.
```
Comment (EpicEric) [inline RELEASE_PROCESS.md]: "a release of of" -> "a release of"

File: .ci-scripts/release.bash [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+chmod 600 $HOME/.netrc
+
+git config --global user.name 'Ponylang Main Bot'
+git config --global user.email 'ponylang.main@gmail.com'
+git config --global push.default simple
+
+# Gather expected arguments.
+if [ $# -lt 1 ]
+thenx
```
Comment (EpicEric) [inline .ci-scripts/release.bash]: thenx -> then

**Comment by SeanTAllen** [comment]
I'm going to be making further changes to this. I spent some time thinking about where things could go wrong and how to make them easily restartable without having to spin up the environment in a special fashion.

**Comment by SeanTAllen** [comment]
@EpicEric can you give another review, I made some reasonably large changes to make this more robust to failures and being able to be restarted at more points in the process.

File: RELEASE_PROCESS.md (hunk 1 above)
Comment (SeanTAllen) [reply RELEASE_PROCESS.md]: fixed

File: .ci-scripts/release.bash (hunk 2 above)
Comment (SeanTAllen) [reply .ci-scripts/release.bash]: fixed

File: .ci-scripts/release/start-a-release.bash [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+# update CHANGELOG for new entries
+echo -e "\e[34mAdding new 'unreleased' section to CHANGELOG.md"
+changelog-tool unreleased -e
+
+# commit changelog and push to master
+echo -e "\e[34mCommiting CHANGELOG.md change"
+git add CHANGELOG.md
+git commit -m "Add unreleased section to CHANGELOG post ${VERSION} release
+
+[skip ci]"
```
Comment (EpicEric) [inline .ci-scripts/release/start-a-release.bash]: Some CI will still run if this isn't in the first line of the commit message. But it's not a big deal, you can leave this as-is.

Comment (SeanTAllen) [reply .ci-scripts/release/start-a-release.bash]: I'll adjust it just to be sure

Comment (SeanTAllen) [reply .ci-scripts/release/start-a-release.bash]: updated.

File: .ci-scripts/release/announce-a-release.bash [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+# Upload release notes
+echo -e "\e[34mUploading release notes..."
+result=$(curl -X POST "https://api.github.com/repos/${GITHUB_REPOSITORY}/releases" \
+  -H "Content-Type: application/x-www-form-urlencoded" \
+  -u "${GITHUB_ACTOR}:${GITHUB_TOKEN}" \
+  --data "${json}")
+
+rslt_scan=$(echo "${result}" | jq -r '.id')
+if [ "$rslt_scan" != null ]
+then
```
Comment (Theodus) [inline .ci-scripts/release/announce-a-release.bash]: These `then`s should be moved to the previous line to be consistent with the rest of the shell scripts (2 others below)

Comment (SeanTAllen) [reply .ci-scripts/release/announce-a-release.bash]: Fixed.

**Pull request merged**